data_steward/analytics/cdr_ops/rdr_export_qc.py

```diff
@@ -669,3 +670,111 @@
                    project_id=project_id,
                    combined_pfmhh_concepts=DROP_PFMHH_CONCEPTS)
 execute(client, query)
+
+# # Check that the Question and Answer Concepts in the old_map_short_codes tables are not paired with 0-valued concept_identifiers
+
+# According to this [ticket](https://precisionmedicineinitiative.atlassian.net/browse/DC-2488), Question and Answer concepts that are identified in the `old_map_short_codes` table should not be paired with 0-valued concept_identifiers after the RDR dataset is cleaned. These concept identifiers include the `observation_concept_id` and `observation_source_concept_id` fields.
+
+# ### Question Codes
+
+# Check the question codes
+tpl = JINJA_ENV.from_string("""
+WITH question_codes AS (
+  SELECT
+    pmi_code
+  FROM `{{project_id}}.{{sandbox_dataset}}.old_map_short_codes` 
+  WHERE type = 'Question'
+)
+SELECT
+  qc.pmi_code, COUNT(*) invalid_id_count
+FROM `{{project_id}}.{{dataset}}.observation` o
+JOIN question_codes qc
+  ON qc.pmi_code = o.observation_source_value
+WHERE (o.observation_source_concept_id = 0
+  OR o.observation_concept_id = 0)
+GROUP BY qc.pmi_code
+ORDER BY 2 DESC
+""")
+query = tpl.render(project_id=project_id, 
+                   dataset=new_rdr,
+                   sandbox_dataset=new_rdr_sandbox)
+df = execute(client, query)
+
+is_success = len(df) == 0
+status_msg = 'Success' if is_success else 'Failure'
+if is_success:
+    display(HTML(
+        f'''
+            <h2>
+                Check Status: <span style="color: {'red' if not is_success else 'green'}">{status_msg}</span>
+            </h2>
+            <p>
+                No 0-valued concept ids found.
+            </p>
+        ''')
+    )    
+else:
+    display(HTML(
+        f'''
+            <h2>
+                Check Status: <span style="color: {'red' if not is_success else 'green'}">{status_msg}</span>
+            </h2>
+            <p>
+                <b>{len(df)}</b> question codes have 0-valued IDs. Report failure back to curation team.
+                Bug likely due to failure in the <code>update_questions_answers_not_mapped_to_omop</code> cleaning rule.
+            </p>
+        ''')
+    )
+    display(df)
+
+# ### Answer Codes
+
+# Check the answer codes
+tpl = JINJA_ENV.from_string("""
+WITH answer_codes AS (
+  SELECT
+    pmi_code
+  FROM `{{project_id}}.{{sandbox_dataset}}.old_map_short_codes` 
+  WHERE type = 'Answer'
+)
+SELECT
+  ac.pmi_code, COUNT(*) invalid_id_count
+FROM `{{project_id}}.{{dataset}}.observation` o
+JOIN answer_codes ac
+  ON ac.pmi_code = o.value_source_value
+WHERE (o.observation_source_concept_id = 0
+  OR o.observation_concept_id = 0)
+GROUP BY ac.pmi_code
+ORDER BY 2 DESC
+""")
```

Comment:
I asked for your update to the code above.  When I run your query with a few extra fields, it is interesting and something we may want to incorporate.  This shows that for some Answers, their questions are not mapped to a non-zero concept_id.  Meaning, this data would be dropped in the clean datasets because both `observation_concept_id` and `observation_source_concept_id` are zero.  But the answer itself is ok.  These will be addressed when the csv file is updated, but I still think this is a useful check.  What do you think about incorporating it?
```sql
WITH answer_codes AS (
  SELECT
    pmi_code
  FROM `{{project_id}}.{{sandbox_dataset}}.old_map_short_codes` 
  WHERE type = 'Answer'
)
SELECT
  ac.pmi_code, o.value_source_value, o.value_source_concept_id, o.value_as_concept_id, o.observation_source_value, o.observation_concept_id, o.observation_source_concept_id,COUNT(*) invalid_id_count
FROM `{{project_id}}.{{dataset}}.observation` o
JOIN answer_codes ac
  ON ac.pmi_code = o.value_source_value
WHERE (o.observation_source_concept_id = 0
  OR o.observation_concept_id = 0)
GROUP BY ac.pmi_code, 2, 3, 4, 5, 6, 7
ORDER BY 8 DESC;
```